### PR TITLE
Better init setup command

### LINKED ISSUE #70: Try to find a way to use native npm install
Currently you don't see anything.
Maybe we should try to find a way to trigger a native install so people will see progress.
A workaround might be to output the command to copy/paste (for now) instead of hidding the install.

### LINKED ISSUE #74: Default package.json as download
Have the default package.json as a download able file from the docs, would be a nice addition.

Maybe with a form that automatically fills in your name and website?

### PR DESCRIPTION
**Pull request by thangngoc89**
This will close #122, close #70, close #74
Ideally, user should type these command to setup a new statinamic project

```
$DIR=your-website && mkdir $DIR && cd $DIR
npm init -y
npm install statinamic
./node_modules/.bin/statinamic setup
# Answer some questions
npm install
```

I also added a `boilerplate` command in npm scripts for unit test.

### CODE REVIEW

**Comment by thangngoc89** [comment]
@MoOx ready for review. How can I make `mkdir` command universal ?

**Comment by MoOx** [comment]
Don't worry, most people will create the folder by hand if they are on windows :) No big deal.
I will review and make comment later.

**Comment by thangngoc89** [comment]
I mean for appveyor.
This is setup test command 

```
mkdir boilerplate-test && cd boilerplate-test && npm init -f && npm install ../ && ./node_modules/.bin/statinamic setup -t
```

**Comment by MoOx** [comment]
https://www.npmjs.com/package/mkdirp have a bin

**Comment by thangngoc89** [comment]
Oops. I just remove `mkdirp` in #126

**Comment by thangngoc89** [comment]
Alright. Add mkdirp back. `boilerplate` can run on appveyor now.

File: src/bin/data/questions.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+import validUrl from "valid-url"
+
+export const defaultAnswers = {
+  name: "Statinamic",
+  homepage: "http://moox.io/statinamic/",
```
Comment (MoOx) [inline src/bin/data/questions.js]: We should not provide any default for name and homepage.

File: src/bin/data/questions.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  CNAME: false,
+  nojekyll: true,
```
Comment (MoOx) [inline src/bin/data/questions.js]: nojekyll should (I think) not be asked and added by default. Enabled by default (already enabled if not provided, per "configurator") and just documented.

File: src/bin/statinamic-setup.js [hunk 3]
```diff
@@ -1,38 +1,71 @@
...
-npmInstallPackage(
-  depsMap,
-  {
-    saveDev: true,
-  },
-  (err) => {
-    if (err) {
-      throw err
-    }
+(function() {
```
Comment (MoOx) [inline src/bin/statinamic-setup.js]: no need to wrap in a function

File: src/bin/data/questions.js (hunk 1 above)
Comment (thangngoc89) [inline src/bin/data/questions.js]: This is for unit test only (-t,  --test option)

File: src/bin/statinamic-setup.js (hunk 3 above)
Comment (thangngoc89) [inline src/bin/statinamic-setup.js]: Ideally yes. But babel will transform async funtion and but it in the top,  before `babel-polyfill' import statement. This cause some reference error.  Maybe add a comment to clarify that?

File: package.json [hunk 4]
```diff
@@ -147,7 +150,9 @@
     "tests": "ava --verbose \"**/__tests__/*.js\"",
     "pretest": "npm run lint",
     "test": "npm run tests && npm run docs",
-    "test-with-coverage": "nyc npm run tests && npm run docs",
+    "test-with-coverage": "nyc npm run tests && npm run docs && npm run boilerplate",
+    "preboilerplate": "rimraf boilerplate-test",
+    "boilerplate": "mkdirp boilerplate-test && cd boilerplate-test && npm init -f && npm install ../ && ./node_modules/.bin/statinamic setup -t",
```
Comment (MoOx) [inline package.json]: can you also add "npm run boilerplate" in "test" ? Btw, maybe rename boilerplate to "test-boilerplate"

Comment (MoOx) [inline package.json]: why npm install .. ? Can't you just use a symlink like for the docs (see lnsf commands)?

**Comment by MoOx** [comment]
Looks good. Just small tweak and that will be great.

File: package.json (hunk 4 above)
Comment (thangngoc89) [inline package.json]: Will change it to test-boilerplate

lnsf won't add an executable bin to node_modules/.bin 

Maybe `npm link ../` for faster test

File: src/bin/data/questions.js (hunk 1 above)
Comment (MoOx) [inline src/bin/data/questions.js]: So please rename defaultAnswers to defaultTestAnswers

File: src/bin/statinamic-setup.js (hunk 3 above)
Comment (MoOx) [inline src/bin/statinamic-setup.js]: Why is polyfill required?
Anyway a comment is better than nothing :)

File: package.json (hunk 4 above)
Comment (MoOx) [inline package.json]: You can call the bin directly with ./src/bin/index.js, but not sure what is the best.

File: src/bin/statinamic-setup.js (hunk 3 above)
Comment (thangngoc89) [inline src/bin/statinamic-setup.js]: Polyfill is required for async... await

File: package.json (hunk 4 above)
Comment (thangngoc89) [inline package.json]: Why are you again `npm link` ? It uses symlink too

**Comment by thangngoc89** [comment]
Should be good by now. 

Changed:

- Use npm link instead of npm install
- Rename to test-boilerplate
- Rename `defaultAnswers` to `defaultTestAnswers`

**Comment by MoOx** [comment]
It seems node 4 test are unhappy. npm 2 issue related to the npm link?

**Comment by MoOx** [comment]
Otherwise, good job!

**Comment by thangngoc89** [comment]
On it.

**Comment by thangngoc89** [comment]
Node 4 is good now.

**Comment by MoOx** [comment]
Will merge after "next".

**Comment by MoOx** [comment]
Conflict so I cherry picked/updated the commit directly in master.

**Pull request closed without merging**